## Assignment 1 Solution

You only need to write lines of code for each question. We will go through comparable code and concepts in the live learning session. The internet is also a great resource when coding (though note that no outside searches are required by the assignment!). If you do incorporate code from the internet, please cite the source within your code (providing a URL is sufficient).

Please bring questions that you cannot work out on your own to office hours, work periods or share with your peers on Slack. We will work with you through the issue.

You will need to install [PLINK](https://zzz.bwh.harvard.edu/plink/) and run the analyses. PLINK is a free, open-source whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner.


#### Question 1: Data inspection

Before fitting any model, it is essential to understand our data. Use R or bash code to answer the following questions about the dataset gwa.qc.A1.fam, gwa.qc.A1.bim and gwa.qc.A1.bed.

(i) Read the .fam file. How many samples does the dataset contain?

```{r,echo=FALSE,message=FALSE, warning=FALSE}
setwd('/Users/macbook/work/Code/Dataset2/')
library(genio)
library(data.table)
library(Matrix)
library(seqminer)
library(ggplot2)
```

```{r,message=FALSE, warning=FALSE}
fam_all <- fread(paste0('gwa.qc.A1.fam'))
N=dim(fam_all)[1];N
```

(ii) What is the 'variable type' of the response variable (i.e.Continuous or binary)?

```{r,echo=FALSE,message=FALSE, warning=FALSE}

print( "Continuous response variable." )

```


(iii) Read the .bim file. How many SNPs does the dataset contain?

```{r,message=FALSE, warning=FALSE}
bim_all<- fread(paste0('gwa.qc.A1.bim'))
M <- nrow(bim_all);M
```

#### Question 2: Allele Frequency Estimation

(i) Load the genotype matrix for SNPs rs1861, rs3813199, rs3128342, and rs11804831 using additive coding. What are the allele frequencies for these four SNPs?

```{r,message=FALSE, warning=FALSE}
indx.SNP=which((bim_all$V2)%in%c('rs1861','rs3813199','rs3128342','rs11804831'))

geno_mat<- readPlinkToMatrixByIndex('gwa.qc.A1', 1:N, indx.SNP)


allele_freq <- apply(geno_mat, 2, function(x) {
  sum(x, na.rm = TRUE) / (2 * sum(!is.na(x)))
})

# Look at the result
allele_freq
```

(ii) What are the minor allele frequencies (MAFs) for these four SNPs?

```{r,message=FALSE, warning=FALSE}
# Convert to MAF
maf <- ifelse(allele_freq > 0.5, 1 - allele_freq, allele_freq)

maf
```


#### Question 3: Hardy–Weinberg Equilibrium (HWE) Test

(i) Conduct the Hardy–Weinberg Equilibrium (HWE) test for all SNPs in the .bim file. Then, load the file containing the HWE p-value results and display the first few rows of the resulting data frame.

```{bash,results='hide',message=FALSE, warning=FALSE}
./plink2 --bfile gwa.qc.A1 --hardy --out gwa_qc_hwe
```


(ii) What are the HWE p-values for SNPs rs1861, rs3813199, rs3128342, and rs11804831?

```{r,message=FALSE, warning=FALSE}
# View HWE results
hwe <- fread("gwa_qc_hwe.hardy")
hwe$P[hwe$ID%in%c('rs1861','rs3813199','rs3128342','rs11804831')]

```


#### Question 4: Genetic Association Test

(i) Conduct a linear regression to test the association between SNP rs1861 and the phenotype. What is the p-value?

```{bash,echo=FALSE,results='hide',message=FALSE, warning=FALSE}
./plink2 --bfile gwa.qc.A1 --freq --out gwa_qc_freq

```

```{r,results='hide',echo=FALSE,message=FALSE, warning=FALSE}
freq <- fread("gwa_qc_freq.afreq")
#freq[which(freq$ID%in%'rs1861'),]
```

```{r,message=FALSE, warning=FALSE}
indx.SNP1=which((bim_all$V2)%in%'rs1861')
geno_SNP<- readPlinkToMatrixByIndex('gwa.qc.A1', 1:N, indx.SNP1)
geno_SNP[geno_SNP==-9]=NA
geno_subset=as.data.frame(cbind(fam_all$V6,geno_SNP))
geno_subset=na.omit(geno_subset)
colnames(geno_subset)=c('Phenotype','Genotype')
AF = mean(geno_subset$Genotype) / 2
# Flip if A1 is major allele
if(AF>0.5){
  geno_subset$Genotype <- 2 - geno_subset$Genotype
}

model <- lm(Phenotype ~ Genotype, data = geno_subset)

# View model summary
summary(model)

print('The p-value is <2e-16')
```

(ii) How would you interpret the beta coefficient from this regression?

```{}
The beta coefficient represents the estimated change in the continuous phenotype for each additional minor allele A (under additive coding) of SNP rs1861. A negative beta suggests the minor allele A is associated with decreased phenotype value.

```

(iii) Plot the scatterplot of phenotype versus the genotype of SNP rs1861. Add the regression line to the plot.

```{r,message=FALSE, warning=FALSE}

ggplot(geno_subset, aes(x=Genotype, y=Phenotype)) +
  geom_point() +
  geom_smooth(method="lm", se=TRUE, color="blue") +
  labs(title="Phenotype vs Additive Genotype (rs1861)",
       x="Genotype (additive coding)",
       y="Phenotype")

```


(iv) Convert the genotype coding for rs1861 to recessive coding.

```{r,message=FALSE, warning=FALSE}
geno_subset$Genotype <- ifelse(geno_subset$Genotype == 2, 1, 0)  
```


(v) Conduct a linear regression to test the association between the recessive-coded rs1861 and the phenotype. What is the p-value?

```{r,message=FALSE, warning=FALSE}
model_rec <- lm(Phenotype ~ Genotype, data=geno_subset)
summary(model_rec)

# To extract and display p-value
summary(model_rec)$coefficients["Genotype", "Pr(>|t|)"]

```

(vi) Plot the scatterplot of phenotype versus the recessive-coded genotype of rs1861. Add the regression line to the plot.

```{r,message=FALSE, warning=FALSE}
ggplot(geno_subset, aes(x=Genotype, y=Phenotype)) +
  geom_point() +
  geom_smooth(method="lm", se=TRUE, color="red") +
  labs(title="Phenotype vs Recessive Genotype (rs1861)",
       x="Genotype (recessive coding)",
       y="Phenotype")

```


(vii) Which model fits better? Justify your answer.

```{}
The additive model fits better because it preserves the linear trend across genotypes (0,1,2) and shows a more significant p-value and better visual fit in the scatterplot. The recessive model loses information by combining heterozygotes with the reference genotype, resulting in worse model fit. The additive model has higher Adjusted R-squared (0.08901) than the recessive model (0.01195), showing better fit. Thus, the additive model is preferred.

```